vnext/proposals/active/keyboard-reconcile-desktop.md

```diff
@@ -0,0 +1,148 @@
+# Reconcile Keyboarding APIs in React Native - desktop
+
+There are several APIs and behaviors around keyboarding introduced in Windows and macOS, some of which match and some don&#39;t. Windows implementation is a bit more complete. The ultimate goal is to reconcile and come up with a singular set of keyboarding APIs and behaviors for RN desktop that can then be proposed for upstreaming into core as well as documented cleanly. This document captures the set of APIs and behaviors and identifies:
+
+- Deltas between implementations in order to reconcile one with the other
+- Deltas in scenarios that need to be completed in both
+- Any other fall out work
+
+## acceptsKeyboardFocus
+
+Property that determines whether this View, Touchable should be focusable with a non-touch input device, eg. receive focus with a hardware keyboard. Setting the property to true will make that View compoonent become a part of tabbing loop.
+
+| **TYPE** | **REQUIRED** | **PLATFORM** | **DEFAULT** |
+| --- | --- | --- | --- |
+| boolean | No | Windows, macOS | false |
+
+React Native core introduced `focusable` that does the same thing as acceptsKeyboardFocus in v0.62. In order to align and make this complete, the work here includes the following:
+- **COMPLETED** : Windows has added `focusable` and marked `acceptsKeyboardFocus` as deprecated in 0.62. 
+- macOS needs to do add `focusable` and mark `acceptsKeyboardFocus` as deprecated. Tracked by - [Issue#498](https://github.com/microsoft/react-native-macos/issues/498)
+- **COMPLETED** : Add focusable to Pressable ([windows Issue#5512](https://github.com/microsoft/react-native-windows/issues/5512), [macOS Issue#500](https://github.com/microsoft/react-native-macos/issues/500)). 
+- `focusable` is already supported in Android and exists in core. There is no hard upstreaming requirement, however – it would be good to implement this for iOS/iPadOS for completion.
+
+## focus, blur, onFocus, onBlur on View
```

Comment:
From @trueadm:
> It's also interesting to know, as we found with the web, if there should be a kind of "blur" event that fires when an actively focused control is removed/hidden. Chrome does this on the web (the only browser that did the last time I tested it). In ReactDOM, we detect this and trigger a set of custom DOM events called "beforeblur" and "afterblur" to emulate and re-create the same experience in all browsers.
This is an important feature of focus management systems, especially in the cases where you want focus to recover in the case when an active control that was focused is removed/hidden. In these cases you typically move focus to a header/sibling, to align with accessibility best standards.

Reply:
Focus management is definitely a much larger system in native frameworks as well. For example, Windows has a whole [FocusManager class](https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.input.focusmanager?view=winrt-19041) with several methods and events including ones that seem equivalent to the `beforeblur` and `afterblur` you have mentioned. 

Not sure how much of all this complexity we want to bring to React Native and how much of this will be reconcilable across platforms.

Reply:
The hardest part about some of those events is that they are required to run synchronously.  Today it would be hard to expose those events to RN in a meaningful way without breaking the thread rules.

Reply:
Something to revisit after Fabric?

Reply:
Sounds fair. I filed an issue to track revisiting this post Fabric - #5706